tests/unit/regression/krrnystrom_unittest.cc

```diff
@@ -0,0 +1,186 @@
+* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*
+* The views and conclusions contained in the software and documentation are those
+* of the authors and should not be interpreted as representing official policies,
+* either expressed or implied, of the Shogun Development Team.
+*/
+
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/kernel/GaussianKernel.h>
+#include <shogun/regression/KRRNystrom.h>
+#include <shogun/regression/KernelRidgeRegression.h>
+#include <shogun/base/some.h>
+#include <gtest/gtest.h>
+
+#ifdef HAVE_CXX11
+
+using namespace shogun;
+
+/**
+ * Test setters are available
+ */
+TEST(KRRNystrom, test_setters)
+{
+	/* Use default constructor */
+	auto krr=some<CKRRNystrom>();
+
+	/* Test setters */
+	krr->set_tau(0.1);
+	krr->set_epsilon(0.1);
+	krr->set_num_rkhs_basis(100);
+}
+
+
+/**
+ * Test 'CKRRNystrom::get_name' method
+ */
+TEST(KRRNystrom, test_get_name)
+{
+	auto krr=some<CKRRNystrom>();
+	EXPECT_EQ(krr->get_name(), "KRRNystrom");
+}
+
+/**
+ * Test the main algorithm by comparison of the alphas and the predictions,
+ * using all columns in the approximation to make sure results are very similar
+ */
+TEST(KRRNystrom, compare_to_KRR_with_all_columns)
+{
+	/* data matrix dimensions */
+	index_t num_vectors=10;
+	index_t num_features=1;
+
+	/* training label data */
+	SGVector<float64_t> lab(num_vectors);
+
+	/* fill data matrix and labels */
+	SGMatrix<float64_t> train_dat(num_features, num_vectors);
+	SGMatrix<float64_t> test_dat(num_features, num_vectors);
+	for (index_t i=0; i<num_vectors; ++i)
+	{
+		/* labels are linear plus noise */
+		lab.vector[i]=i+CMath::normal_random(0, 1.0);
+		train_dat.matrix[i]=i;
+		test_dat.matrix[i]=i;
+	}
+
+	/* training features */
+	auto features=some<CDenseFeatures<float64_t>>(train_dat);
+	auto features_krr=some<CDenseFeatures<float64_t>>(train_dat);
+
+	/* testing features */
+	auto test_features=some<CDenseFeatures<float64_t>>(test_dat);
+
+	/* training labels */
+	auto labels=some<CRegressionLabels>(lab);
+	auto labels_krr=some<CRegressionLabels>(lab);
+
+	/* kernel */
+	auto kernel=some<CGaussianKernel>(features, features, 10, 0.5);
+	auto kernel_krr=some<CGaussianKernel>(features_krr, features_krr, 10, 0.5);
+
+	/* kernel ridge regression and the nystrom approximation */
+	float64_t tau=0.01;
+	auto nystrom=some<CKRRNystrom>(tau, num_vectors, kernel, labels);
+	auto krr=some<CKernelRidgeRegression>(tau, kernel_krr, labels_krr);
+
+	nystrom->train();
+	krr->train();
+
+	SGVector<float64_t> alphas=nystrom->get_alphas();
+	SGVector<float64_t> alphas_krr=krr->get_alphas();
+
+	for (index_t i=0; i<num_vectors; ++i)
+		EXPECT_NEAR(alphas[i], alphas_krr[i], 1E-1);
+
+	auto result=(CRegressionLabels *)nystrom->apply(test_features);
```

Comment:
use `apply_regression` which will directly give you regression labels and you can avoid the cast